pkg/plugins/volume/ceph/size.go

```diff
@@ -0,0 +1,130 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package ceph
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/ceph/go-ceph/rados"
+	"github.com/ceph/go-ceph/rbd"
+	"github.com/go-logr/logr"
+	"github.com/ironcore-dev/libvirt-provider/pkg/api"
+)
+
+func createKeyFile(imageName string, key string) (string, func() error, error) {
+	noOpCleanup := func() error { return nil }
+	file, err := os.CreateTemp("", fmt.Sprintf("%s-*", imageName))
+	if err != nil {
+		return "", noOpCleanup, fmt.Errorf("failed to create temp file: %w", err)
+	}
+	cleanup := func() error {
+		return os.Remove(file.Name())
+	}
+
+	_, err = file.WriteString(key)
+	if err != nil {
+		return "", cleanup, fmt.Errorf("failed to write key to temp file: %w", err)
+	}
+
+	return file.Name(), cleanup, nil
```

Comment:
Where are those return `funcs` are being called? I assume you want to do a `defer cleanup`.

Reply:
Here: 
https://github.com/ironcore-dev/libvirt-provider/blob/1e6ff11e7cd27d426db68bc1f69ff28db919bcf3/pkg/plugins/volume/ceph/size.go#L93C2-L97C5
The authentication file needs to stay until the connection is being closed, therefore the cleanup needs to happen outside of this function.